src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,610 @@
+        public static implicit operator JsonNumber(float value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a Double to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(double value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a SByte to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(sbyte value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a UInt16 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(ushort value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a UInt32 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(uint value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a UInt64 to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(ulong value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a Decimal to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(decimal value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Compares <paramref name="obj"/> to the numeric value of this instance. 
+        /// </summary>
+        /// <param name="obj">The object to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the numeric value of this instance matches <paramref name="obj"/>,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public override bool Equals(object obj) => obj is JsonNumber number && _value == number._value;
+
+        /// <summary>
+        ///   Calculates a hash code of this instance.
+        /// </summary>
+        /// <returns>A hash code for this instance.</returns>
+        public override int GetHashCode() => _value.GetHashCode();
+
+        /// <summary>
+        ///   Compares other JSON number to the numeric value of this instance. 
+        /// </summary>
+        /// <param name="other">The JSON number to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the numeric value of this instance matches <paramref name="other"/>,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public bool Equals(JsonNumber other) => _value == other._value;
```

Comment:
What if I pass in `null` to this method?

Reply:
I will fix it.